tests/rust/pipe-test/pop_wait/client.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+//======================================================================================================================
+// Imports
+//======================================================================================================================
+
+use ::anyhow::Result;
+use ::demikernel::{
+    demi_sgarray_t,
+    runtime::types::demi_opcode_t,
+    LibOS,
+    QDesc,
+    QToken,
+};
+use ::std::slice;
+use std::time::Duration;
+
+//======================================================================================================================
+// Structures
+//======================================================================================================================
+
+/// Pipe client
+pub struct PipeClient {
+    /// Underlying libOS.
+    libos: LibOS,
+    /// Underlying pipe descriptor.
+    pipeqd: Option<QDesc>,
```

Comment:
I don't see any reason for this to be an Option value. Looks like it is set in the new function and removed in drop.

Reply:
Indeed, it does not. I'll remove the `Option` value before merging.